main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UnlockCommand.cs

```
@@ -40,33 +42,45 @@ public static bool Unlock (VersionControlItemList items, bool test)
 				return false;
 			if (test)
 				return true;
-			
-			new UnlockWorker (items).Start();
+
+			new UnlockWorker (items).Start ();
 			return true;
 		}
 
-		private class UnlockWorker : VersionControlTask 
+		private class UnlockWorker : VersionControlTask
 		{
 			VersionControlItemList items;
-						
-			public UnlockWorker (VersionControlItemList items) {
+
+			public UnlockWorker (VersionControlItemList items)
+			{
 				this.items = items;
 			}
-			
-			protected override string GetDescription() {
+
+			protected override string GetDescription ()
+			{
 				return GettextCatalog.GetString ("Unlocking...");
 			}
-			
+
 			protected override void Run ()
 			{
-				foreach (VersionControlItemList list in items.SplitByRepository ())
-					list[0].Repository.Unlock (Monitor, list.Paths);
-				
-				Gtk.Application.Invoke ((o, args) => {
-					VersionControlService.NotifyFileStatusChanged (items);
-				});
-				Monitor.ReportSuccess (GettextCatalog.GetString ("Unlock operation completed."));
+				try {
+					foreach (VersionControlItemList list in items.SplitByRepository ()) {
+						try {
+							list [0].Repository.Unlock (Monitor, list.Paths);
+						} catch (Exception ex) {
+							Monitor.ReportError (ex.Message, null);
+							LoggingService.LogError ("Unlock operation failed", ex);
+							return;
```

we already catch in the outer try/catch block, returning here would suppress the error message alert. I'd keep the original behavior for now, just log the error additionally in the outer catch (line 81)

---

Changed.